src/api/roles.coffee

```diff
@@ -0,0 +1,248 @@
+Channel = require('../model/channels').Channel
+Client = require('../model/clients').Client
+logger = require 'winston'
+authorisation = require './authorisation'
+utils = require '../utils'
+
+
+###
+# Roles is a virtual API; virtual in the sense that it is not linked
+# to a concrete roles collection.
+#
+# Rather it an abstraction of the 'allow' field on Channels and 'roles' on Clients,
+# providing a mechanism for setting up allowed permissions.
+###
+
+
+filterRolesFromChannels = (channels, allClients) ->
+  roles = {}
```

Comment:
I'm guessing an object is used initially to make sure we don't have duplicates. Perhaps a comment to this effect would be useful to the reader.